### DISCUSSION TITLE
How can I access ContentItem properties in an Email workflow activity

### DISCUSSION THREAD

**Discussion by akim-green** [OrchardCMS/OrchardCore#7453] [Q&A]
I can see in the fields of the email workflow event page that it supports liquid, but I can't seem to get it to work. I set up the workflow to send an email once an item is published, but I cannot access the ContentItem's properties in the body of the email. It just render whatever I type. I would like to be able to render {{Model.ContentItem.Address}}, is that possible?

**Comment by netwavebe** [comment]
ContentItem can be retrieved via the workflow input-property, like `{{ Workflow.Input["ContentItem"] }}`

**Comment by netwavebe** [comment]
I don't know what `Address` is, but try `{{ Workflow.Input["ContentItem"].Content.Address.Text }}`.

**Reply by akim-green** [reply]
Thanks for your response but that doesn't work for me.. I have a simple "UserProfile" Content Item with a username, password and some other properties, I am trying to send an email when a profile is update but it doesn't work.

This is the state of the workflow after execution:
`        "Recipients": {
          "Expression": "{{Workflow.Input[\"ContentItem\"].Content.Email.Text}}"
        }`

**Reply by netwavebe** [reply]
IIRC, a user is not a content item as this would bring security issues.

Are you listening to any of the user events in your workflow, like `User Updated Event`? In that case `{{ Workflow.Input["User"] }}` gives you the class defined in `OrchardCore.Users.Models.User`.

If not, what is the starting event in your workflow?

**Reply by akim-green** [reply]
UserProfile is a custom ContentType I defined in the CMS.. I was trying to user the `ContentItem Update` Event which does fire off, but then an exception gets thrown in the Email service because I cannot access the Content Item "email" property in the Email Task

**Reply by jptissot** [reply]
What is the structure of your content item ?

**Reply by akim-green** [reply]
@jptissot  Here is the structure of the content item: 

`{"ContentItemId":"440n1jppfsva63wpcsbk6qaamm","ContentItemVersionId":"4x7bg0c4eqdxc1413s1t0pyp59","ContentType":"UserProfile","DisplayText":null,"Latest":true,"Published":true,"ModifiedUtc":"2020-11-03T15:23:17.0005728Z","PublishedUtc":"2020-11-03T15:23:17.0846677Z","CreatedUtc":"2020-11-03T15:23:17.0005728Z","Owner":"Akim","Author":"Akim","UserProfile":{"Username":{"Text":"kingakim"},"Email":{"Text":"kingakim@hotmail.com"},"Role":{"Text":"storeAdmin"}},"StoreIds":{"ContentItems":[{"ContentItemId":"4akcvan6kc63j3vcpwgnsqt9dg","ContentItemVersionId":null,"ContentType":"StoreId","DisplayText":null,"Latest":false,"Published":false,"ModifiedUtc":"2020-11-03T15:23:16.9711538Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"Akim","StoreId":{"StoreId":{"Text":"190450"}}}]},"Token":{"ContentItems":[]}}`

What I would ideally like to do is be able to trigger the email workflow event when the bag part is modified, rather than a general update, if this is not possible using workflows, can you please guide as to how I can send an email programmatically using orchard services. Thanks
